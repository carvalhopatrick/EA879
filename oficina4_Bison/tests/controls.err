not implemented: table -- code will be parsed but will have no effect (line 2)
error: syntax error (line 2)
