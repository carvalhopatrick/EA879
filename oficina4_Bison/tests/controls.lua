-- examples from https://en.wikibooks.org/wiki/Lua_Programming/Tables
array = {5, "text", {"more text", 463, "even more text"}, "more text"}
array = {
	"text";
	{
		"more text, in a nested table";
		1432
	};
	true -- booleans can also be in tables
}
dictionary = {
	"text";
	text = "more text";
	654;
	[8] = {}; -- an empty table
	func = function() print("tables can even contain functions!") end;
	["1213"] = 1213
}
dictionary = {number = 6}
print(dictionary["number"])

dictionary = {["number"] = 6}
print(dictionary.number)
nested_table = {
	[6] = {
		func = function() print("6") end
	}
}
nested_table[6].func()

array = {5, 2, 6, 3, 6}
for index, value in next, array do
	print(index .. ": " .. value + 1)
end

local object = {}
function object.func(self, arg1, arg2)
	print(arg1, arg2)
end

local object = {}
function object:func(arg1, arg2)
	print(arg1, arg2)
end

-- examples from https://en.wikibooks.org/wiki/Lua_Programming/Statements

local number = 0
while number < 10 do
	print(number)
	number = number + 1 -- Increase the value of the number by one.
end

local number = 0
repeat
	print(number)
	number = number + 1
until number >= 10

for number = 0, 9, 1 do
	print(number)
end

for n = 1, 2, 0.1 do
	print(n)
	if n >= 1.5 then
		break -- Terminate the loop instantly and do not repeat.
	end
end

