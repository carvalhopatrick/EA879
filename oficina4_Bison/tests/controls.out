<unimplemented/not executed>
<unimplemented/not executed>
